### PR TITLE
refactor: Calling org in tools is changed from bundle to grpc

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Calling org in tools is changed from bundle to grpc 

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    Calling org in tools is changed from bundle to grpc           |
| 🇨🇳 中文    |     调用 org 从 bundle 改为 grpc         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: cmd/pipeline/main.go [hunk 1]
```diff
@@ -21,10 +21,12 @@ import (
 	"github.com/erda-project/erda/internal/tools/pipeline/conf"
 	"github.com/erda-project/erda/pkg/common"
 
+	_ "github.com/erda-project/erda-infra/providers/grpcclient"
```
Comment (sfwn) [inline cmd/pipeline/main.go:24]: put under comment

File: internal/core/messenger/notifygroup/provider.go [hunk 2]
```diff
@@ -95,7 +95,7 @@ func init() {
 	servicehub.Register("erda.core.messenger.notifygroup", &servicehub.Spec{
 		Services:             pb.ServiceNames(),
 		Types:                pb.Types(),
-		OptionalDependencies: []string{"service-register"},
+		OptionalDependencies: []string{"service-register", "audit"},
```
Comment (sfwn) [inline internal/core/messenger/notifygroup/provider.go]: put into dependencies

File: internal/core/openapi/legacy/api/apis/uc/uc_user_update_userinfo.go [hunk 3]
```diff
@@ -91,6 +92,7 @@ func updateUserInfo(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	orgClient := r.Context().Value("org").(org.Interface)
```
Comment (sfwn) [inline internal/core/openapi/legacy/api/apis/uc/uc_user_update_userinfo.go:95]: define a const for "org" as other provider dependencies

File: internal/core/openapi/legacy/api/apis/openapi/component_protocol/render.go [hunk 4]
```diff
@@ -144,6 +145,9 @@ func legacyProtocolRender(w http.ResponseWriter, r *http.Request) {
 	ctx := context.Background()
 	ctx1 := context.WithValue(ctx, protocol.GlobalInnerKeyCtxBundle.String(), ctxBdl)
 
+	orgClient := r.Context().Value("org").(org.Interface)
```
Comment (sfwn) [inline internal/core/openapi/legacy/api/apis/openapi/component_protocol/render.go:148]: org

File: internal/core/openapi/legacy/auth/user.go [hunk 5]
```diff
@@ -192,11 +197,14 @@ func (u *User) GetOrgInfo(orgHeader, domainHeader string) (orgID uint64, err err
 		return 0, nil
 	}
 	// query org info
-	org, err := u.bundle.GetOrg(orgName)
+	// query org info
```
Comment (sfwn) [inline internal/core/openapi/legacy/auth/user.go:200]: duplicated comment

File: internal/core/openapi/legacy/api/apis/uc/uc_user_update_userinfo.go (hunk 3 above)
Comment (sfwn) [reply internal/core/openapi/legacy/api/apis/uc/uc_user_update_userinfo.go:95]: use const defines at internal/core/openapi/legacy/component-protocol/protocol.go

File: internal/core/openapi/legacy/proxy/ws/ws.go [hunk 6]
```diff
@@ -91,6 +94,7 @@ func (r *ReverseProxyWithCustom) ServeHTTP(rw http.ResponseWriter, req *http.Req
 		}
 	}()
 	spec := api.API.Find(req)
+	req = req.WithContext(context.WithValue(req.Context(), "org", r.org))
```
Comment (sfwn) [inline internal/core/openapi/legacy/proxy/ws/ws.go:97]: this ctx already have Org passed by LoginServer

File: internal/core/openapi/legacy/proxy/http/http.go [hunk 7]
```diff
@@ -54,6 +57,7 @@ func (r *ReverseProxyWithCustom) ServeHTTP(rw http.ResponseWriter, req *http.Req
 	}()
 	spec := api.API.Find(req)
 	req.Header.Add(util.UserInfoDesensitizedHeader, strconv.FormatBool(spec.NeedDesensitize))
+	req = req.WithContext(context.WithValue(req.Context(), "org", r.org))
```
Comment (sfwn) [inline internal/core/openapi/legacy/proxy/http/http.go:60]: bad literal `org`

File: internal/core/openapi/openapi-ng/auth/ory-kratos/check.go [hunk 8]
```diff
@@ -131,3 +138,16 @@ func (p *provider) getScope(r *http.Request, userID string) (uint64, error) {
 	}
 	return orgID, nil
 }
+
+func (p *provider) GetOrg(IdOrName string) (*orgpb.Org, error) {
```
Comment (sfwn) [inline internal/core/openapi/openapi-ng/auth/ory-kratos/check.go:142]: why do not use p.org.getorg.
I think impl of proto and bundle is equal.

File: internal/core/openapi/openapi-ng/auth/uc-session/user.go [hunk 9]
```diff
@@ -33,7 +33,7 @@ func (p *provider) GetUserInfo(rw http.ResponseWriter, r *http.Request) {
 	if len(session) > 0 {
 		r = r.WithContext(context.WithValue(r.Context(), "session", session))
 	}
-	user := auth.NewUser(p.Redis)
+	user := auth.NewUser(p.Redis, p.Org)
```
Comment (sfwn) [inline internal/core/openapi/openapi-ng/auth/uc-session/user.go:36]: use pkg level var directly

**Review by sfwn — CHANGES_REQUESTED** [review]

File: cmd/pipeline/main.go (hunk 1 above)
Comment (littlejiancc) [reply cmd/pipeline/main.go:24]: done

File: internal/core/messenger/notifygroup/provider.go (hunk 2 above)
Comment (littlejiancc) [reply internal/core/messenger/notifygroup/provider.go]: done

File: internal/core/openapi/legacy/api/apis/openapi/component_protocol/render.go (hunk 4 above)
Comment (littlejiancc) [reply internal/core/openapi/legacy/api/apis/openapi/component_protocol/render.go:148]: done

File: internal/core/openapi/legacy/auth/user.go (hunk 5 above)
Comment (littlejiancc) [reply internal/core/openapi/legacy/auth/user.go:200]: done

File: internal/core/openapi/legacy/api/apis/uc/uc_user_update_userinfo.go (hunk 3 above)
Comment (littlejiancc) [reply internal/core/openapi/legacy/api/apis/uc/uc_user_update_userinfo.go:95]: done

File: internal/core/openapi/openapi-ng/auth/uc-session/user.go (hunk 9 above)
Comment (littlejiancc) [reply internal/core/openapi/openapi-ng/auth/uc-session/user.go:36]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**